libavcodec/rkmppdec.c

```diff
@@ -352,6 +351,18 @@ static int rkmpp_retrieve_frame(AVCodecContext *avctx, AVFrame *frame)
         goto fail;
     }
 
+    if (decoder->eos_reached && !mpp_frame_get_eos(mppframe)) {
+        if (mppframe)
+            mpp_frame_deinit(&mppframe);
+        usleep(10000);
+        goto retry_get_frame;
+    }
+    if (mppframe && mpp_frame_get_eos(mppframe)) {
+        av_log(avctx, AV_LOG_DEBUG, "EOS frame found\n");
+        ret = AVERROR_EXIT;
+        goto fail;
+    }
+
```

Comment:
That return code is not appropriate here. Once again i have already upstreamed those changes and if you feel that a few things should be changed, you need to send those patches to ffmpeg upstream, as the reference code is there now :)